backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java

```diff
@@ -1967,6 +1972,10 @@ public CurrencyConversionContext getCurrencyConversionCtx(@NonNull final org.com
 		final ForexContractRef forexContractRef = InvoiceDAO.extractForeignContractRef(invoice);
 		if (forexContractRef != null)
 		{
+			Optional.ofNullable(forexContractRef.getForexContractId())
+					.map(id -> forexContractServiceLoader.get().getById(id))
+					.ifPresent(forexContract -> forexContract.validateSectionCode(SectionCodeId.ofRepoIdOrNull(invoice.getM_SectionCode_ID())));
```

Comment:
loud thinking: why validate it here and why not when the invoice is generated?

Reply:
bc I want to make sure the section code is right when the document gets posted, not created.. as the section code can be changed at any given time.